--- a/Makefile	2015-08-04 21:23:52.407411740 +0200
+++ b/Makefile	2015-08-04 21:24:11.815441176 +0200
@@ -143,15 +143,14 @@
 		$(MAKECMDPREFIX)$(MAKECMD) -C $(CFGDIR) clean
 
 install_bin:
-		mkdir -p $(SBINDIR)
-		install -m 755 $(EXENAME) $(SBINDIR)
-		$(STRIP) $(SBINDIR)/$(EXENAME)
+		mkdir -p $(INSTALL_PREFIX)$(SBINDIR)
+		install -m 755 $(EXENAME) $(INSTALL_PREFIX)$(SBINDIR)
 ifeq ($(SGW_SUPPORT),1)
 		$(MAKECMDPREFIX)if [ -e $(SBINDIR)/$(SGW_POLICY_SCRIPT) ]; then \
-			cp -f files/$(SGW_POLICY_SCRIPT) $(SBINDIR)/$(SGW_POLICY_SCRIPT).new; \
+			cp -f files/$(SGW_POLICY_SCRIPT) $(INSTALL_PREFIX)$(SBINDIR)/$(SGW_POLICY_SCRIPT).new; \
 			echo "Policy routing script was saved as $(SBINDIR)/$(SGW_POLICY_SCRIPT).new"; \
 		else \
-			cp -f files/$(SGW_POLICY_SCRIPT) $(SBINDIR)/$(SGW_POLICY_SCRIPT); \
+			cp -f files/$(SGW_POLICY_SCRIPT) $(INSTALL_PREFIX)$(SBINDIR)/$(SGW_POLICY_SCRIPT); \
 		fi
 endif
 
@@ -169,22 +168,16 @@
 		@echo per default.
 		@echo can be found at files/olsrd.conf.default.lq
 		@echo ==========================================================
-		mkdir -p $(ETCDIR)
-		$(MAKECMDPREFIX)if [ -e $(CFGFILE) ]; then \
-			cp -f files/olsrd.conf.default.lq $(CFGFILE).new; \
-			echo "Configuration file was saved as $(CFGFILE).new"; \
-		else \
-			cp -f files/olsrd.conf.default.lq $(CFGFILE); \
-		fi
+		mkdir -p $(INSTALL_PREFIX)$(ETCDIR)
 		@echo -------------------------------------------
 		@echo Edit $(CFGFILE) before running olsrd!!
 		@echo -------------------------------------------
 		@echo Installing manpages $(EXENAME)\(8\) and $(CFGNAME)\(5\)
 ifneq ($(MANDIR),)
-		mkdir -p $(MANDIR)/man8/
-		cp files/olsrd.8.gz $(MANDIR)/man8/$(EXENAME).8.gz
-		mkdir -p $(MANDIR)/man5/
-		cp files/olsrd.conf.5.gz $(MANDIR)/man5/$(CFGNAME).5.gz
+		mkdir -p $(INSTALL_PREFIX)$(MANDIR)/man8/
+		cp files/olsrd.8.gz $(INSTALL_PREFIX)$(MANDIR)/man8/$(EXENAME).8.gz
+		mkdir -p $(INSTALL_PREFIX)$(MANDIR)/man5/
+		cp files/olsrd.conf.5.gz $(INSTALL_PREFIX)$(MANDIR)/man5/$(CFGNAME).5.gz
 endif
 ifneq ($(RCDIR),)
 		cp $(RCFILE) $(RCDIR)/olsrd
@@ -233,16 +226,16 @@
 endif
 
 libs:
-		$(MAKECMDPREFIX)set -e;for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(LIBDIR);done
+		$(MAKECMDPREFIX)set -e;for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(INSTALL_PREFIX)$(LIBDIR);done
 
 libs_clean clean_libs:
-		-for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(LIBDIR) clean;rm -f lib/$$dir/*.so lib/$$dir/*.dll;done
+		-for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(INSTALL_PREFIX)$(LIBDIR) clean;rm -f lib/$$dir/*.so lib/$$dir/*.dll;done
 
 libs_install install_libs:
-		$(MAKECMDPREFIX)set -e;for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(LIBDIR) install;done
+		$(MAKECMDPREFIX)set -e;for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(INSTALL_PREFIX)$(LIBDIR) install;done
 
 libs_uninstall uninstall_libs:
-		$(MAKECMDPREFIX)set -e;for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(LIBDIR) uninstall;done
+		$(MAKECMDPREFIX)set -e;for dir in $(SUBDIRS);do $(MAKECMD) -C lib/$$dir LIBDIR=$(INSTALL_PREFIX)$(LIBDIR) uninstall;done
 		rmdir -p $(LIBDIR) || true
 
 #
