--- a/Makefile.inc	2020-02-18 14:04:10.350884951 +0000
+++ b/Makefile.inc	2020-02-18 14:05:59.176026673 +0000
@@ -20,6 +20,9 @@
 # shows full compiler/linker calls if activated
 VERBOSE ?= 0
 
+# when non-zero will exclude rpath linker option
+NORPATH ?= 0
+
 SANITIZE_ADDRESS ?= 0
 
 ifeq ($(VERBOSE),0)
@@ -176,8 +179,10 @@
 else 
   LDFLAGS +=	-Wl,-export-dynamic 
 endif
+ifeq ($(NORPATH),0)
 LDFLAGS +=	-Wl,-rpath,$(LIBDIR)
 endif
+endif
 # LDFLAGS for all 
 LDFLAGS +=	-Wl,--warn-common
 # See above at "-ffunction-sections" for an explanation (and why it is disabled).
